assets/js/base/components/shipping-calculator/address.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { __ } from '@wordpress/i18n';
+import Button from '@woocommerce/base-components/button';
+import { ShippingCountryInput } from '@woocommerce/base-components/country-input';
+import { ShippingCountyInput } from '@woocommerce/base-components/county-input';
+import TextInput from '@woocommerce/base-components/text-input';
+import { useState } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const ShippingCalculatorAddress = ( { address: initialAddress, onUpdate } ) => {
+	const [ address, setAddress ] = useState( initialAddress );
+
+	return (
+		<div className="wc-block-shipping-calculator-address">
+			<ShippingCountryInput
+				className="wc-block-shipping-calculator-address__input"
+				label={ __(
+					'Country / Region',
+					'woo-gutenberg-products-block'
+				) }
+				value={ address.country }
+				onChange={ ( newValue ) =>
+					setAddress( {
+						...address,
+						country: newValue,
+						county: '',
+					} )
+				}
+			/>
+			<ShippingCountyInput
+				className="wc-block-shipping-calculator-address__input"
+				country={ address.country }
+				label={ __( 'County', 'woo-gutenberg-products-block' ) }
```

Comment:
If we want the default to be generic, this should probably be `State / County` like in core.

Reply:
Good point. Changed in 576f46e94897abb68c0adacf895ddda9ce40b766.